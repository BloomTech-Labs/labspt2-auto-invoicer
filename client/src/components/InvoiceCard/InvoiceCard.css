.invoice-card {
  width: 250px;
  height: 180px;
  border: 2px solid gray;
  border-radius: 5px;
  margin: 10px;
  padding: 10px;
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex-direction: column;
}

.invoice-card h3 {
  font-size: 2rem;
  font-weight: bold;
}

.invoice-card p {
  font-size: 1.4rem;
  text-align: justify;
}

.status-unpaid{
  border-color: rgb(224, 55, 55);
  box-shadow: 0 0 5px red;
}

.status-paid {
  border-color: green;
  box-shadow: 0 0 5px green;
}
