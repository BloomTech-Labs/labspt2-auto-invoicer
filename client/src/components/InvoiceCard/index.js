import React from "react";
import PropTypes from "prop-types";

// import css here
import "./InvoiceCard.css";

class InvoiceCard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
        invoiceDueDate: this.props.invoice.invoiceDueDate,
        amountPaid: this.props.invoice.amountPaid,
        total: this.props.invoice.total,
        paid: this.props.invoice.paid
      }
    };
  
  lateChecker = date => {
    let year = date.slice(11, 15);
    let month = date.slice(4, 7);
    let day = date.slice(8, 10);
    let monthConvertedToNumber = 0;
    let months = [
      "Jan",
      "Feb",
      "Mar",
      "Apr",
      "May",
      "Jun",
      "Jul",
      "Aug",
      "Sep",
      "Oct",
      "Nov",
      "Dec"
    ];
    let monthInNumberForm = [
      "01",
      "02",
      "03",
      "04",
      "05",
      "06",
      "07",
      "08",
      "09",
      "10",
      "11",
      "12"
    ];
    for (let i = 0; i < months.length; i++) {
      if (months[i] === month) {
        monthConvertedToNumber = monthInNumberForm[i];
      }
    }
    let dateInNumberForm = parseInt(year + monthConvertedToNumber + day, 10);
    return dateInNumberForm;
  };
  visualClass = () => {
    if (this.state.amountPaid >= this.state.total) {
      return "status-paid invoice-card";
    } else if (
      this.lateChecker(Date()) >
      this.lateChecker(String(this.state.invoiceDueDate))
    ) {
      return "status-late invoice-card";
    } else {
      return "status-unpaid invoice-card";
    }
  };
  render() {
    const {
      invoiceNumber,
      invoiceNotes,
      invoiceDueDate,
      amountPaid,
      total,
      paid
    } = this.props.invoice;
    return (
      <article className={this.visualClass()}>
        <h3>Invoice Number: {invoiceNumber}</h3>
        <p>
          <u>Invoice Due Date:</u>
          <br /> {invoiceDueDate}
        </p>
        <p>
          <u>Total:</u>
          <br /> ${total}
        </p>
        <p>
          <u>Amount Paid:</u>
          <br /> ${amountPaid}
        </p>
        <p>
          <u>Notes:</u>
          <br /> {invoiceNotes}
        </p>
      </article>
    );
  }
}

InvoiceCard.propTypes = {
  invoice: PropTypes.shape({
    invoiceNumber: PropTypes.number,
    invoiceDueDate: PropTypes.string,
    total: PropTypes.string,
    amountPaid: PropTypes.string,
    invoiceNotes: PropTypes.string,
    paidStatus: PropTypes.string
  })
};

export default InvoiceCard;
