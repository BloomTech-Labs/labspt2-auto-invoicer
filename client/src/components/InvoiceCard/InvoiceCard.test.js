import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import InvoiceCard from "../InvoiceCard";

describe("<InvoiceCard />", () => {
  it("renders Invoice Card", () => {
    const { asFragment } = render(
      <InvoiceCard
        invoice={{
          invoiceNumber: 0,
          text: "testing invoice card",
          paidStatus: "paid"
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
