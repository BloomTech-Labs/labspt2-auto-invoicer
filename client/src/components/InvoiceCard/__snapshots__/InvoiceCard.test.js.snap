// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InvoiceCard /> renders Invoice Card 1`] = `
<DocumentFragment>
  <article
    class="invoice-card paid"
  >
    <h3>
      0
    </h3>
    <p>
      testing invoice card
    </p>
  </article>
</DocumentFragment>
`;
